### PR TITLE
moving "is_integrally_closed" to the category setup.

### PR DESCRIPTION
**Pull request by fchapoton**
This is moving a method from the auld Ring class to the category setup.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [x] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: src/sage/categories/rings.py [hunk 1]
```diff
@@ -423,6 +423,28 @@ def is_integral_domain(self, proof=True) -> bool:
...
+                sage: R.is_integrally_closed()
+                False
+                sage: R
+                Order of conductor 2 generated by 2*a in Number Field in a with defining polynomial x^2 + 189*x + 394
+                sage: S = K.maximal_order(); S
+                Maximal Order generated by a in Number Field in a with defining polynomial x^2 + 189*x + 394
+                sage: S.is_integrally_closed()
+                True
+            """
+            return False
```
Comment (vincentmacri) [inline src/sage/categories/rings.py]: I'm not too familiar with the Sage category setup, but why isn't the default implementation to raise a `NotImplementedError`? It looks like that was the previous behaviour.

Comment (fchapoton) [reply src/sage/categories/rings.py]: yes, indeed, that's another possibility and that was the previous behaviour. We can also decide to say "False" if there is no specific reason known for the property in question to hold. I have no strong opinion.

Comment (vincentmacri) [reply src/sage/categories/rings.py]: > yes, indeed, that's another possibility and that was the previous behaviour. We can also decide to say "False" if there is no specific reason known for the property in question to hold. I have no strong opinion.

I feel that it's better to raise an error than report a potentially incorrect result.

Comment (vincentmacri) [inline src/sage/categories/rings.py]:
```suggestion
            raise NotImplementedError
```
And change the docstring accordingly.

**Review by vincentmacri — CHANGES_REQUESTED** [review]

File: src/sage/categories/rings.py (hunk 1 above)
Comment (fchapoton) [reply src/sage/categories/rings.py]: voila, voila

**Review by vincentmacri — APPROVED** [review]
LGTM

**Comment by mantepse** [comment]
I'm overriding the needs-review bot.

**Pull request merged**